src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -132,13 +181,21 @@ internal void Update(double value, ReadOnlySpan<KeyValuePair<string, object?>> t
     internal int Snapshot()
     {
         this.batchSize = 0;
-        var indexSnapshot = Math.Min(this.metricPointIndex, this.maxMetricPoints - 1);
-        if (this.outputDelta)
+        if (this.OutputDelta)
         {
-            this.SnapshotDelta(indexSnapshot);
+            if (this.reclaimMetricPoints)
+            {
+                this.SnapshotDeltaWithMetricPointReclaim();
+            }
+            else
+            {
+                var indexSnapshot = Math.Min(this.metricPointIndex, this.maxMetricPoints - 1);
+                this.SnapshotDelta(indexSnapshot);
```

Comment:
`reclaimMetricPoints` is set to true under a lock though always read outside of the lock. What happens if a SnapshotDelta is in flight while another thread sets `reclaimMetricPoints` to true?

Reply:
> What happens if a SnapshotDelta is in flight while another thread sets reclaimMetricPoints to true?

The thread running `Collect` would continue to execute `SnapshotDelta` and the next `Collect` run would switch to executing `SnapshotDeltaWithMetricPointReclaim`. Essentially, the reclaiming behavior would start only for subsequent `Snapshot` calls (post the execution of the in-flight `SnapshotDelta`.